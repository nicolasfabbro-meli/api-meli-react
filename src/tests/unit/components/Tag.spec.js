import React from 'react';
import Component from '../../../components/Tag';

describe('Tag', () => {
  const props = {
    children: '__TEST__',
    handleClick: jest.fn(),
    handleClose: jest.fn(),
  }

  it('Should match snapshot', () => {
    const component = shallow(<Component {...props} />);
    expect(escapeSnapshot(component)).toMatchSnapshot();
  });

  it('Should call handleClick method', () => {
    const component = shallow(<Component {...props} />);
    component.find('.tag span').simulate('click');
    expect(props.handleClick).toHaveBeenCalled();
  });

  it('Should call handleClose method', () => {
    const component = shallow(<Component {...props} />);
    component.find('[name="close-outline"]').simulate('click');
    expect(props.handleClose).toHaveBeenCalled();
  });
});
