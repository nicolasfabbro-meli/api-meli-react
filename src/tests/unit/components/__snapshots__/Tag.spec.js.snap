// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Should match snapshot 1`] = `
"<span className='tag'>
  <span onClick={[Function: onClick]}>
    __TEST__
  </span>
  <ion-icon name='close-outline' onClick={[Function: onClick]} />
</span>"
`;
